single_kernel_mongo/state/app_peer_state.py

```diff
@@ -165,3 +173,40 @@ def external_connectivity(self, value: bool) -> None:
             raise ValueError(
                 f"'external-connectivity' must be a boolean value. Provided: {value} is of type {type(value)}"
             )
+
+    @property
+    def database(self) -> str:
+        """Database tag for mongos."""
+        return self.relation_data.get(AppPeerDataKeys.database.value, "mongos-database")
+
+    @database.setter
+    def database(self, value: str):
+        """Sets database tag in databag."""
+        self.update({AppPeerDataKeys.database.value: value})
+
+    @property
+    def extra_user_roles(self) -> set[str]:
+        """extra_user_roles tag for mongos."""
+        return set(
+            self.relation_data.get(
+                AppPeerDataKeys.extra_user_roles.value,
+                "default",
+            ).split(",")
+        )
+
+    @extra_user_roles.setter
+    def extra_user_roles(self, value: set[str]):
+        """Sets extra_user_roles tag in databag."""
+        roles_str = ",".join(value)
+        self.update({AppPeerDataKeys.extra_user_roles.value: roles_str})
+
+    @property
+    def expose_external(self) -> ExposeExternal:
+        """The value of the expose-external flag."""
+        if not self.relation:
+            return ExposeExternal.UNKNOWN
+        return ExposeExternal(self.relation_data.get(AppPeerDataKeys.expose_external.value, ""))
+
+    @expose_external.setter
+    def expose_external(self, value: ExposeExternal):
+        self.update({AppPeerDataKeys.expose_external.value: value.value})
```

Comment:
is this property for both VM + K8s?

seems like it is just for k8s, but I would love it to work for VM too if possible